File: src/main/Java/boj_step9/boj2501.java
import java.io.*;
import java.util.StringTokenizer;

public class boj2501 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int n = Integer.parseInt(tokenizer.nextToken());
        int k = Integer.parseInt(tokenizer.nextToken());
        int count= 0;
        for (int i = 1; i <= n; i++)
            if (n % i == 0)
                if (++count == k) {
                    writer.append(String.valueOf(i));
                    break;
                }
        if (count < k) writer.append("0");
        writer.close();
    }
}

File: src/main/boj_step9/boj2501.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val (n, k) = r.readLine().split(" ").map { it.toInt() }
    val factors = mutableListOf<Int>()
    repeat(n) {
        if (n % (it + 1) == 0) factors.add(it + 1)
    }
    if(k > factors.size) w.write("0")
    else w.write(factors[k - 1].toString())
    w.close()
}